packages/block-library/src/cover/shared.js

```diff
@@ -73,12 +73,16 @@ export function attributesFromMedia( setAttributes, dimRatio ) {
 			mediaType = media.type;
 		}
 
+		const isDark = getCoverIsDark( dimRatio, averageBackgroundColor?.hex );
```

Comment:
It looks like `attributesFromMedia` is only used when changing the media. And if you are _always_ setting the `customOverlayColor` to the `averageBackgroundColor`, we can skip the `compositeSourceOver` operation. It will _always_ be the same as the `averageBackgroundColor` because you're overlaying the same color on top of itself now.

Reply:
> Am I not doing that by providing the 3rd argument?

It's not quite the same. Overlaying a 50% opaque red over white results in pink (current behavior), but a 50% opaque red over red results in red (correct behavior).

https://github.com/WordPress/gutenberg/pull/54054/files#diff-98e8fbed8e10d45fee8dfcb0373d733aa1335f7cbf0a5dd883739b86fae0fd7eR187-R191